### PR TITLE
[MOSIP-31132] Added limit for wrong attempts in OTP login

### PR DESCRIPTION
**Pull request by kaifk468**

### CODE REVIEW

File: signup-service/src/main/java/io/mosip/signup/dto/RegistrationTransaction.java [hunk 1]
```diff
@@ -58,6 +57,11 @@ public void increaseAttempt() {
         this.lastRetryAt = LocalDateTime.now(ZoneOffset.UTC);
     }
 
+    public void increaseVerificationAttempt() {
+        this.verificationAttempts += 1;
+        this.lastRetryAt = LocalDateTime.now(ZoneOffset.UTC);
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/dto/RegistrationTransaction.java:62]: Why is lastRetryAt updated here?

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java [hunk 2]
```diff
@@ -101,6 +101,9 @@ public class RegistrationService {
     @Value("${mosip.signup.challenge.resend-attempt}")
     private int resendAttempts;
 
+    @Value("${mosip.signup.verification.attempt}")
```
Comment (ase-101) [inline signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:104]: Rename the key to `mosip.signup.challenge.verification-attempt`

File: signup-service/src/main/java/io/mosip/signup/dto/RegistrationTransaction.java (hunk 1 above)
Comment (kaifk468) [reply signup-service/src/main/java/io/mosip/signup/dto/RegistrationTransaction.java:62]: removed

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java (hunk 2 above)
Comment (kaifk468) [reply signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java]: renamed

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java [hunk 3]
```diff
@@ -101,6 +101,9 @@ public class RegistrationService {
     @Value("${mosip.signup.challenge.resend-attempt}")
     private int resendAttempts;
 
+    @Value("${mosip.signup.challenge.verification-attempt}")
```
Comment (balaji-alluru) [inline signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:104]: Please raise a pr to mosip config adding this change to both camdec-dev, and release-1.4.1-ES branches.

File: signup-service/src/main/java/io/mosip/signup/util/ErrorConstants.java [hunk 4]
```diff
@@ -17,6 +17,7 @@ public class ErrorConstants {
     public static final String GENERATE_CHALLENGE_FAILED ="generate_challenge_failed";
     public static final String UNKNOWN_ERROR="unknown_error";
     public static final String TOO_MANY_ATTEMPTS="too_many_attempts";
+    public static final String TOO_MANY_VERIFY_ATTEMPTS="too_many_verify_attempts";
```
Comment (balaji-alluru) [inline signup-service/src/main/java/io/mosip/signup/util/ErrorConstants.java:20]: add this error msg in artifactory branches release-1.4.1-ES, develop, camdgc.

**Review by ase-101 — APPROVED** [review]

File: signup-service/src/main/java/io/mosip/signup/util/ErrorConstants.java (hunk 4 above)
Comment (kaifk468) [reply signup-service/src/main/java/io/mosip/signup/util/ErrorConstants.java:20]: PR of develop  
https://github.com/mosip/artifactory-ref-impl/pull/741

Comment (kaifk468) [reply signup-service/src/main/java/io/mosip/signup/util/ErrorConstants.java:20]: PR of camdgc 
https://github.com/mosip/artifactory-ref-impl/pull/742

File: signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java (hunk 3 above)
Comment (kaifk468) [reply signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:104]: PR of camdgc-dev https://github.com/mosip/mosip-config/pull/5990

Comment (kaifk468) [reply signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java:104]: PR of config for camdgc-dev

**Pull request merged**